<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <link style="text/css" rel="stylesheet" href="/Style/css/bioinfo.css" />
    <link href="/Style/css/page_theme.css" rel="stylesheet" style="text/css"/>
   <link href="/protea/protea.css" rel="stylesheet" style="text/css"/>
    <script type="text/javascript" src="http://code.jquery.com/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="/libs/jquery.history.js"></script>
    <!--<script type="text/javascript" src="/scripts/bioinfo.js"></script>-->
    <script type="text/javascript" src="/scripts/bioinfo.js"></script>
   
   <title>Bonsai  :: Bioinformatics Software Server</title>
    <script type="text/javascript">
        var i_am_old_ie = false;
    </script>
    <!--[if LT IE  9]>
    <script type="text/javascript">
            i_am_old_ie = true;
    </script>
    <![endif]-->
  </head>
  <body>
   
   
    <div class="frametitle">
   <h1 id="title">protea :: protein coding gene prediction</h1>                 
   </div>

   <div id="center_sup">
        <div class="theme-border" style="display:none"></div>
   <div id="link_home" style="display:inline-block"><a href="/" class="text_onglet"><img src="/Style/icon/home_w.png" alt="home_general"/></a></div>
   <div class="tabs" id="menu_central" style="display:inline-block"><?php include("menu_central.txt")?></div>
   </div>
   <div id="main">
    <div id="center">

    <h2>PFAM family PF07974 [<a href="http://pfam.sanger.ac.uk/family/PF07974">description</a>]</h2>
    <ul>
      <li>3 homologous coding sequences: <a href="examples/PF07974.fa">fasta</a></li>
      <li>Average identity percentage: 44.3 %</li>
      <li>Predicted amino acid sequences</li>
    </ul>
    <pre>
         CLUSTAL W (1.83) multiple sequence alignment


         Q19267_CAEEL/158-187      CF---GKGSCHGDGSREGSGKCKCETG----YTGNLC
         Q86G85_PSEIC/635-670      CRSPENNEICSGNG-QCVCGQCMCNSDDDRHYSGKYC
         O97702_CANFA/508-543      CSPREGQPACSQRG-ECLCGQCVCHSSDFGKITGKYC
                                   *    .:  *   * .  .*:* *.:.     :*: *
    </pre>
    <ul>
      <li>Multiple alignment deducted from amino acid sequences
	alignment</li>
    </ul>
    <pre>
         CLUSTAL multiple alignment produced by protea version 0.04
      
         Q19267_CAEEL/158-187      UGUuuu---------ggaaaaggauccUGUCAUGGAGAUGGAAGCCGCGAAGGCAGUGGA
         Q86G85_PSEIC/635-670      UGCcggucaccugaaaacaacgaaaucUGCAGUGGAAACGGA---CAAUGUGUAUGUGGA
         O97702_CANFA/508-543      UGCagcccccgggagggccagcccgccUGCAGCCAGCGGGGC---GAGUGCCUGUGUGGC
	                           ***                *      ***          **              ****

         Q19267_CAEEL/158-187      AAGUGUAAAUGUGAGACUGGA------------UAUACUGGAAAUCUAUGC
         Q86G85_PSEIC/635-670      CAAUGUAUGUGUAACUCUGACGAUGACCGCCACUAUAGUGGCAAAUACUGC
         O97702_CANFA/508-543      CAAUGUGUCUGCCAUAGCAGUGACUUUGGCAAGAUCACGGGCAAGUACUGC
	                            * ***   **  *                      *  ** **    ***
    </pre>






    <h2>PFAM family PF00608 [<a href="http://pfam.sanger.ac.uk/family/PF00608">description</a>]</h2>
    <ul>
      <li>4 homologous coding sequences: <a href="examples/PF00608.fa">fasta</a></li>
      <li>Average identity percentage: 43.5 %</li>
      <li>Predicted amino acid sequences</li>
    </ul>
    <pre>
         CLUSTAL W (1.83) multiple sequence alignment


         FIB1_ADE41/244-273      NNSLSLVITSPLKVINSMLAVGVNPPFTIT
         FIBP_ADE04/129-158      LNTLVSAFGSGLGLSGSALAVQLASPLTFD
         FIB1_ADE40/327-356      TLNVNANTSKGLAIENNSLVVKLGNGLRFD
         FIBP_ADEB3/213-242      DNKTVVKAGPGLTITNQALTVATGNGLQVN
                                   .        * : .. *.*     : .       
    </pre>
    <ul>
      <li>Multiple alignment deducted from amino acid sequences
	alignment</li>
    </ul>
    <pre>
         CLUSTAL multiple alignment produced by protea version 0.04

         FIB1_ADE41/244-273        AACAACUCUCUCUCCUUAGUCAUUACUUCUCCUUUAAAAGUUAUUAAUUCUAUGUUAGCC
         FIBP_ADE04/129-158        UUGAAUACAUUAGUCUCAGCUUUUGGCUCAGGUUUAGGACUCAGUGGCAGCGCCCUGGCA
         FIB1_ADE40/327-356        ACAUUAAACGUCAAUGCCAAUACCAGCAAAGGUCUUGCUAUUGAAAAUAACUCACUAGUU
         FIBP_ADEB3/213-242        GAUAAUAAAACUGUCGUUAAGGCUGGGCCCGGACUUACAAUAACUAAUCAAGCUCUUACU
                                                                     *     *              *    
         FIB1_ADE41/244-273        GUUGGGGUUAACCCGCCUUUUACCAUCACU
         FIBP_ADE04/129-158        GUACAGUUAGCCUCUCCACUUACAUUUGAU
         FIB1_ADE40/327-356        GUUAAGCUAGGAAACGGUCUUCGCUUUGAU
         FIBP_ADEB3/213-242        GUUGCUACCGGGAACGGCCUUCAGGUCAAC
                                   **                 **    *    
    </pre>



    





    <h2>Non coding family: RFAM family RF00455 [<a href="http://rfam.sanger.ac.uk/family/RF00455">description</a>]</h2>
    <ul>
      <li>4 non-coding sequences: <a href="examples/RF00455.fa">fasta</a></li>
      <li>Average identity percentage: 68.4 %</li>
    </ul>

</div></div><!-- bloc droit-->

<?php require("../lib.inc")?>
<?php footer("PROTEA","Protea", "protea@lifl.fr","2013"); ?>
<script src="/protea/js/script.js" type="text/javascript">
</body>                                        
</html>        
